deploy/examples/statefulset-manual-sidecar.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: example-statefulset
+  labels:
+    app: example-app
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: example-app
+  serviceName: example-service
+  template:
+    metadata:
+      labels:
+        app: example-app
+    spec:
+      containers:
+        - name: example-app
+          image: jaegertracing/vertx-create-span:operator-e2e-tests
+          ports:
+            - containerPort: 8080
+              protocol: TCP
+        - name: jaeger-agent
+          image: jaegertracing/jaeger-agent:1.17.0
+          imagePullPolicy: IfNotPresent
+          ports:
+            - containerPort: 5775
+              name: zk-compact-trft
+              protocol: UDP
+            - containerPort: 5778
+              name: config-rest
+              protocol: TCP
+            - containerPort: 6831
+              name: jg-compact-trft
+              protocol: UDP
+            - containerPort: 6832
+              name: jg-binary-trft
+              protocol: UDP
+            - containerPort: 14271
+              name: admin-http
+              protocol: TCP
+          args:
+            - --reporter.grpc.host-port=dns:///jaeger-collector-headless.ephemeral-roles:14250
```

Comment:
This might reserve another name, or at least a comment, saying that this implies an instance named `jaeger` on the namespace `ephemeral-roles`. It would probably be better to fully qualify it, like, `...ephemeral-roles.svc.cluster.local`

Reply:
Great catch, thanks! I accidentally left that in from my testing, I'll change it up to reflect the example :+1: